src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification.Classifiers
+{
+    internal class DiscardSyntaxClassifier : AbstractSyntaxClassifier
+    {
+        public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(
+            typeof(DiscardDesignationSyntax),
+            typeof(DiscardPatternSyntax),
+            typeof(LambdaExpressionSyntax),
+            typeof(IdentifierNameSyntax));
+
+        public override void AddClassifications(
+           Workspace workspace,
+           SyntaxNode syntax,
+           SemanticModel semanticModel,
+           ArrayBuilder<ClassifiedSpan> result,
+           CancellationToken cancellationToken)
+        {
+            if (syntax.IsKind(SyntaxKind.DiscardDesignation) || syntax.IsKind(SyntaxKind.DiscardPattern))
+            {
+                result.Add(new ClassifiedSpan(syntax.Span, ClassificationTypeNames.Keyword));
+                return;
+            }
+
+            var symbolInfo = semanticModel.GetSymbolInfo(syntax, cancellationToken);
+
+            switch (syntax)
+            {
+                case LambdaExpressionSyntax _:
+                    ClassifyLambdaParameter(symbolInfo, result);
+                    break;
+
+                default:
+                    ClassifySymbol(syntax, symbolInfo, result);
+                    break;
+            };
+        }
+
+        private void ClassifyLambdaParameter(SymbolInfo symbolInfo, ArrayBuilder<ClassifiedSpan> result)
+        {
+            // classify lambda parameters of the forms 
+            // (int _, int _) => ... 
+            // or 
+            // (_, _) => ...
+            // which aren't covered by TryClassifySymbol
+
+            if (!(symbolInfo.Symbol is IMethodSymbol symbol))
+            {
+                return;
+            }
+
+            foreach (var parameter in symbol.Parameters)
```

Comment:
would it be simpler to just have the classifier classify ParameterSyntax and just filter to parameters named `"_"`?

Reply:
Wouldn't i basically have to re-roll similar code as it is written here because i need to differentiate between `(_) => 0` and `(_,_) => 0` where only the second case is discarding, meaning i would have to inspect the parent and either go by the `IParameterSymbol.IsDiscard` API again or count parameters in the parent that have an underscore similar to the code in the [lambda binder](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs#L198-L201)

Reply:
you would just do:

`if the parameter name is _ and the parmaeter symbol IsDicard, then classify as a discard`.

Reply:
yeah but to get the parameter symbol i need the expression syntax or rather the `IMethodSymbol`. Calling `semanticModel.GetSymbolInfo(parameterSyntax, cancellationToken);` doesn't return any symbol.

Reply:
call `semanticModel.GetDeclaredSymbol(parameterSyntax)` in that case :)

Reply:
that works :)